setup.py

```diff
@@ -47,10 +47,10 @@
 ]
 
 lint_requires = [
-    "black==23.7.0",
-    "codespell[tomli]==2.2.5",
-    "ruff==0.0.280",
-    "yamllint==1.32.0",
+    "black>=23.7.0,<24.0.0",
+    "codespell[tomli]>=2.2.5,<3.0.0",
+    "ruff==0.0.282",
+    "yamllint>=1.32.0,<2.0.0",
```

Comment:
Maybe we can use `~=` for those deps? Not sure if it is supported in the `setup.py`

Reply:
`~=` sometimes gets weird with renovate, so I'm preferring not right now until they fix it.